protocols/kad/src/protocol.rs

```diff
@@ -346,6 +323,31 @@ pub enum KadResponseMsg {
     },
 }
 
+impl From<KadRequestMsg> for proto::Message {
+    fn from(kad_msg: KadRequestMsg) -> Self {
+        req_msg_to_proto(kad_msg)
+    }
+}
+impl From<KadResponseMsg> for proto::Message {
+    fn from(kad_msg: KadResponseMsg) -> Self {
+        resp_msg_to_proto(kad_msg)
+    }
+}
+impl TryFrom<proto::Message> for KadRequestMsg {
+    type Error = io::Error;
+
+    fn try_from(message: proto::Message) -> Result<Self, Self::Error> {
+        proto_to_req_msg(message)
+    }
+}
+impl TryFrom<proto::Message> for KadResponseMsg {
+    type Error = io::Error;
+
+    fn try_from(message: proto::Message) -> Result<Self, Self::Error> {
+        proto_to_resp_msg(message)
+    }
+}
```

Comment:
Is it worth having this code? As far as I can see, you are still calling all of them manually in the codec, right? I don't think we gain anything from having these then and I'd rather not have these 20 LoC and instead call the functions they are delegating too directly from the codec.

Reply:
That would mean we have to define both `InCodec` and `OutCodec` structs, with different `Encoder` & `Decode` trait impls since we can't rely on just switching the positions of `KadResponseMsg` & `KadRequestMsg` on the same type.

Reply:
Ah you are generic over these! Sorry didn't see that, that is actually quite clean. Nicely done :)